### PR TITLE
Add support to export reimbursable expenses as Expenses to QBO

### PR DESCRIPTION
**Pull request by Sravanksk**

### CODE REVIEW

File: apps/quickbooks_online/tasks.py [hunk 1]
```diff
@@ -453,6 +453,118 @@ def create_cheque(expense_group, task_log_id):
...
+    else:
+        return
+
+    general_settings = WorkspaceGeneralSettings.objects.get(workspace_id=expense_group.workspace_id)
+    try:
+        qbo_credentials = QBOCredential.objects.get(workspace_id=expense_group.workspace_id)
+
+        qbo_connection = QBOConnector(qbo_credentials, expense_group.workspace_id)
+
+        if general_settings.auto_map_employees and general_settings.auto_create_destination_entity:
```
Comment (ashwin1111) [inline apps/quickbooks_online/tasks.py:504]: if you are adding support for CCC, add a condition to create only for `PERSONAL` expenses

File: apps/quickbooks_online/tasks.py [hunk 2]
```diff
@@ -453,6 +453,118 @@ def create_cheque(expense_group, task_log_id):
...
+            create_or_update_employee_mapping(expense_group, qbo_connection, general_settings.auto_map_employees)
+
+        with transaction.atomic():
+            __validate_expense_group(expense_group, general_settings)
```
Comment (ashwin1111) [inline apps/quickbooks_online/tasks.py:508]: Add check for default credit card vendor incase you add support for CCC

File: apps/quickbooks_online/tasks.py [hunk 3]
```diff
@@ -453,6 +453,118 @@ def create_cheque(expense_group, task_log_id):
...
+
+            qbo_expense_object = QBOExpense.create_qbo_expense(expense_group)
+
+            qbo_expense_line_item_objects = QBOExpenseLineitem.create_qbo_expense_lineitems(expense_group)
+
+            created_qbo_expense = qbo_connection.post_qbo_expense(qbo_expense_object, qbo_expense_line_item_objects)
+
+            load_attachments(qbo_connection, created_qbo_expense['Purchase']['Id'], 'Purchase', expense_group)
```
Comment (ashwin1111) [inline apps/quickbooks_online/tasks.py]: Can you bring this at last, after marking as success in task log and expense group.

Comment (ashwin1111) [reply apps/quickbooks_online/tasks.py]: Please update this for existing journal entry, bills, etc ..,

File: apps/quickbooks_online/urls.py [hunk 4]
```diff
@@ -37,6 +38,8 @@
     path('bills/trigger/', BillScheduleView.as_view()),
     path('checks/', ChequeView.as_view()),
     path('checks/trigger/', ChequeScheduleView.as_view()),
+    path('expenses/', QBOExpenseView.as_view()),
```
Comment (ashwin1111) [inline apps/quickbooks_online/urls.py:41]: I don't think this endpoint is needed

File: apps/quickbooks_online/views.py [hunk 5]
```diff
@@ -487,6 +488,54 @@ def post(self, request, *args, **kwargs):
         )
 
 
+class QBOExpenseView(generics.ListCreateAPIView):
```
Comment (ashwin1111) [inline apps/quickbooks_online/views.py:491]: Don't think this is needed

File: apps/quickbooks_online/models.py [hunk 6]
```diff
@@ -401,6 +401,124 @@ def create_cheque_lineitems(expense_group: ExpenseGroup):
...
+                destination_type='ACCOUNT',
+                source__value=category,
+                workspace_id=expense_group.workspace_id
+            ).first()
+
+            class_id = get_class_id_or_none(expense_group, lineitem)
+
+            customer_id = get_customer_id_or_none(expense_group, lineitem)
+
+            qbo_expense_lineitem_object, _ = QBOExpenseLineitem.objects.update_or_create(
```
[the remark this answers is not in the document]
Comment (ashwin1111) [reply apps/quickbooks_online/models.py:504]: TODO: get green from Shwetabh for just create()

File: apps/quickbooks_online/tasks.py (hunk 2 above)
Comment (Sravanksk) [reply apps/quickbooks_online/tasks.py:508]: doing only for `reimbursable` now

**Review by Shwetabhk — APPROVED** [review]

**Pull request merged**